### PR TITLE
SCUMM: COMI (Demo): properly handle demo music and transitions

### PR DESCRIPTION
**Pull request by AndywinXp**
These modifications make so that COMI Demo music is handled mostly in the same way as it'shandled in the full game.
As a bonus, iMUSE transition crossfades between tracks automatically work!

### CODE REVIEW

File: engines/scumm/imuse_digi/dimuse_music.cpp [hunk 1]
```diff
@@ -343,6 +365,53 @@ void IMuseDigital::playComiMusic(const char *songName, const imuseComiTable *tab
...
+		if (table->filename[0] == 0) {
+			fadeOutMusic(60);
+			return;
+		}
+		if (getCurMusicSoundId() == table->soundId)
+			return;
+		if ((table->attribPos != 0) &&
+			(table->attribPos == _comiDemoStateMusicTable[_curMusicState].attribPos)) {
+			fadeOutMusicAndStartNew(table->fadeOutDelay, table->filename, table->soundId);
+		}
```
Comment (aquadran) [inline engines/scumm/imuse_digi/dimuse_music.cpp:406]: formatting

File: engines/scumm/imuse_digi/dimuse_music.cpp [hunk 2]
```diff
@@ -299,7 +321,7 @@ void IMuseDigital::playComiMusic(const char *songName, const imuseComiTable *tab
 		fadeOutMusic(120);
 		return;
 	}
-
+
```
Comment (aquadran) [inline engines/scumm/imuse_digi/dimuse_music.cpp:324]: formatting

Comment (AndywinXp) [reply engines/scumm/imuse_digi/dimuse_music.cpp:324]: I'm sorry, I don't understand, that function is unchanged from the previous pull request.

File: engines/scumm/imuse_digi/dimuse_music.cpp (hunk 1 above)
Comment (aquadran) [reply engines/scumm/imuse_digi/dimuse_music.cpp:406]: line should be "} else {"

Comment (AndywinXp) [reply engines/scumm/imuse_digi/dimuse_music.cpp]: Sorry, fixed; thanks for your patience!

**Pull request merged**